lib/application.js

```diff
@@ -141,7 +143,12 @@ app.lazyrouter = function lazyrouter() {
       strict: this.enabled('strict routing')
     });
 
-    this._router.use(query(this.get('query parser fn')));
+    var parser = this.get('query parser fn');
+    if (this.defaultExtendedQueryParser && parser && parser.name === 'parseExtendedQueryString') {
```

Comment:
We could probably just skip checking the parser and just warning on the tracking property, especially since `parser.name` will not always be the same for those who are using manglers to pack their app, which typically alters the function name.

Reply:
done, removed the parser name check, and solely depended on the tracker variable.